### PR TITLE
Upgrade to go 1.20

### PR DESCRIPTION
**Pull request by jrockway**
CORE-1519

### CODE REVIEW

File: src/internal/sdata/sql.go [hunk 1]
```diff
@@ -38,9 +38,7 @@ func (m *SQLTupleWriter) Flush() error {
 	// flatten list of Tuple
 	var values Tuple
 	for r := range m.buf {
-		for c := range m.buf[r] {
-			values = append(values, m.buf[r][c])
-		}
+		values = append(values, m.buf[r]...)
```
Comment (brendoncarroll) [inline src/internal/sdata/sql.go:41]: Lol this seems unrelated, but good to merge nonetheless.

**Comment by brendoncarroll** [comment]
A few more places to consider, there might be good reasons not to upgrade some of them right now.

- [ ] `./src/integrations/connectors/snowflake/Dockerfile`
- [ ] `./src/server/cmd/pachd/signals_unix.go`
- [ ] `etc/testing/circle/rootless_test.sh`
- [ ] `etc/testing/circle/start-minikube.sh`
- [ ] `etc/testing/kafka/Dockerfile`
- [ ] `etc/testing/kafka/go.mod`
- [ ] `etc/proto/Dockerfile`
- [ ] `etc/proto/protoc-gen-zap/go.mod`
- [ ] `etc/testing/spout/Dockerfile`
- [ ] `etc/testing/spout/go.mod`

File: src/internal/sdata/sql.go (hunk 1 above)
Comment (jrockway) [reply src/internal/sdata/sql.go:41]: Oh, the linter demands it now.

**Comment by jrockway** [comment]
Please take another look.  Explicitly requiring 1.20 for everything looks pretty OK.  I also ran "make proto" locally since I think CI doesn't test that, and it works fine.

**Review by brendoncarroll — APPROVED** [review]
LGTM

**Pull request merged**